src/languageService/kustoLanguageService.ts

```diff
@@ -839,19 +832,30 @@ class KustoLanguageService implements LanguageService {
     }
 
     getFormattedCommandsInDocumentV2(document: ls.TextDocument, rangeStart?: number, rangeEnd?: number): {
-        formattedCommands: string[], rangeBeforeFormat?: ls.Range
+        formattedCommands: string[], originalRange?: ls.Range
      } {
         const script = this.parseDocumentV2(document);
 
         const commands = this
             .toArray<k2.CodeBlock>(script.Blocks)
             .filter((command) => {
-                const newLineLength = "\r\n".length;
                 if (!command.Text || command.Text.trim() == '') return false;
                 if (rangeStart == null || rangeEnd == null) return true;
+
+                // calculate command end position without \r\n.
```

Comment:
why do we need this logic? what happens if we don't "cut" the newlines?

Reply:
Without this change, these selections:
![image](https://user-images.githubusercontent.com/8294298/88091537-0593ac00-cb44-11ea-8907-6c0a7a48b2f1.png)
![image](https://user-images.githubusercontent.com/8294298/88091587-1e9c5d00-cb44-11ea-9655-6846ad938862.png)

Will also prettify the first command.